src/Reduce.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+declare(strict_types=1);
+
+namespace IterTools;
+
+class Reduce
+{
+    /**
+     * Reduces given collection like array_reduce() function.
+     *
+     * But unlike array_reduce(), it works with all iterable types.
+     *
+     * @template T
+     * @param iterable<mixed> $data
+     * @param callable        $reducer
+     * @param T               $carry
+     *
+     * @return T
+     */
+    public static function toValue(iterable $data, callable $reducer, $carry = null)
```

Comment:
The third parameter here, `$carry` is exposing an implementation detail. I think a better abstraction for this from the client point of view is as the "initial value." So, what do you think about renaming this to `$initialValue` and then initialize a `$carry` variable prior to the foreach loop?

Reply:
Totally agree. Done.